agents/tests/test_agent.py

```diff
@@ -205,3 +211,89 @@ def mock_urlopen(request, timeout=5):
     assert msg in contents, (
         'Expected %s in logs, got:\n%s' % (msg, contents)
     )
+
+
+def test_post_data_success(monkeypatch):
```

Comment:
Just a comment: I'm surprised `monkeypatch` works in Python 2.6, for some reason, I've convinced myself it doesn't 🤔